src/main/java/net/hydromatic/lookml/ObjectHandler.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed to the LookML Authors under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this
+ * work for additional information regarding copyright ownership.
+ * The LookML Authors license this file to you under the Apache
+ * License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License.  You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.  See the License for the specific
+ * language governing permissions and limitations under the
+ * License.
+ */
+package net.hydromatic.lookml;
+
+import java.util.function.Consumer;
+
+/** Handles events that occur while parsing a LookML object.
+ *
+ * @see ListHandler */
+public interface ObjectHandler {
+  /** Adds a numeric property. */
+  ObjectHandler number(String propertyName, Number value);
+
+  /** Adds a string property. */
+  ObjectHandler string(String propertyName, String value);
```

Comment:
Sure, it's an option. If the set of property types was large or unbounded having a generic property value class seems to win out. (We did that with `interface PropertyHandler` in #6 because after validation each property has its own type.) If there is a small fixed set of property types (as here) then a method for each type of property seems simpler.